users/groups/service_test.go

```diff
@@ -267,11 +265,15 @@ func TestUpdateGroup(t *testing.T) {
 	}
 
 	for _, tc := range cases {
-		repoCall := pRepo.On("Evaluate", context.Background(), "group", mock.Anything).Return(nil)
+		repoCall := pRepo.On("CheckAdmin", context.Background(), mock.Anything).Return(nil)
 		repoCall1 := gRepo.On("Update", context.Background(), mock.Anything).Return(tc.response, tc.err)
 		expectedGroup, err := svc.UpdateGroup(context.Background(), tc.token, tc.group)
 		assert.True(t, errors.Contains(err, tc.err), fmt.Sprintf("%s: expected %s got %s\n", tc.desc, tc.err, err))
 		assert.Equal(t, tc.response, expectedGroup, fmt.Sprintf("%s: expected %v got %v\n", tc.desc, tc.response, expectedGroup))
+		if tc.err == nil {
+			repoCall.Parent.AssertCalled(t, "CheckAdmin", context.Background(), mock.Anything)
+			repoCall1.Parent.AssertCalled(t, "Update", context.Background(), mock.Anything)
```

Comment:
`AssertCalled` does not do assertion itself, but **returns a bool value** that needs to be checked. Either way, it looks like this `if err == nil` is irrelevant, since these two methods are always called.

Reply:
Actually no. Since we have an `"update group name with invalid token"` it would fail on identify

Reply:
Chaning it to something like this

```go
if tc.err == nil {
	if !repoCall.Parent.AssertCalled(t, "CheckAdmin", context.Background(), mock.Anything) {
		assert.Fail(t, fmt.Sprintf("CheckAdmin was not called on %s", tc.desc))
	}
	if !repoCall1.Parent.AssertCalled(t, "Update", context.Background(), mock.Anything) {
		assert.Fail(t, fmt.Sprintf("Update was not called on %s", tc.desc))
	}
}
```

Reply:
You can use constructs:
```go
c := repoCall.Parent.AssertCalled(t, "Save", context.Background(), mock.Anything)
assert.True(t, c, fmt.Sprintf("Save was not called on %s", tc.desc))
```
instead of `if`.